### PR TITLE
Fixes #5930

### LINKED ISSUE #5930: "not" queries in molfiles get inverted
**Describe the bug**
"Not" queries get turned into the atom that I'm looking to exclude. 

**To Reproduce**

```
In [1]: from rdkit import Chem                                                                                                                                                                                                
In [2]: text = """ 
  ...:      RDKit          2D 
  ...:   
  ...:   0  0  0  0  0  0  0  0  0  0999 V3000 
  ...: M  V30 BEGIN CTAB 
  ...: M  V30 COUNTS 1 0 0 0 0 
  ...: M  V30 BEGIN ATOM 
  ...: M  V30 1 "NOT [N]" -2.742857 0.057143 0.000000 0 
  ...: M  V30 END ATOM 
  ...: M  V30 END CTAB 
  ...: M  END 
  ...: $$$$ 
  ...: """                                                                                                                                                                                                                   

In [3]: mol = Chem.MolFromMolBlock(text)                                                                                                                                                                                      
In [4]: print(Chem.MolToMolBlock(mol, forceV3000=True))                                                                                                                                                                                        


     RDKit          2D

  0  0  0  0  0  0  0  0  0  0999 V3000
M  V30 BEGIN CTAB
M  V30 COUNTS 1 0 0 0 0
M  V30 BEGIN ATOM
M  V30 1 N -2.742857 0.057143 0.000000 0
M  V30 END ATOM
M  V30 END CTAB
M  END

```

**Expected behavior**
I'd expect to get back the "not N". This also happens if I use the `forceV3000` flag, it must be happening on read. If I change the input atom name from "NOT [N]" to "NOT [N,C]", then everything works as expected and the output matches the input.

**Configuration (please complete the following information):**
 - RDKit version: 2022.09.3
 - Are you using conda? no, a local Schrodinger build.

### PR DESCRIPTION
**Pull request by greglandrum**
This requires a few small logic changes to allow single-atom list queries to be written.

### CODE REVIEW

**Comment by greglandrum** [comment]
@d-b-w : you may also want to look at this

**Review by ricrogz — APPROVED** [review]
LGTM

File: Code/GraphMol/FileParsers/MolFileWriter.cpp [hunk 1]
```diff
@@ -1382,7 +1383,7 @@ std::string MolToMolBlock(const ROMol &mol, bool includeStereo, int confId,
   if (trwmol.needsUpdatePropertyCache()) {
     trwmol.updatePropertyCache(false);
   }
-  if (kekulize) {
+  if (kekulize && mol.getNumBonds()) {
```
Comment (bp-kelley) [inline Code/GraphMol/FileParsers/MolFileWriter.cpp:1386]: Is this related to this PR?

Comment (greglandrum) [reply Code/GraphMol/FileParsers/MolFileWriter.cpp:1386]: yes: the tests include a molecule with a single aromatic atom. That led to errors when generating a CTAB

File: Code/GraphMol/FileParsers/MolFileWriter.cpp [hunk 2]
```diff
@@ -400,6 +400,7 @@ const std::string AtomGetMolFileSymbol(
             static_cast<ATOM_EQUALS_QUERY *>(atom->getQuery())->getVal() ==
                 1))) {
         res = "A";
+        queryListAtoms.set(atom->getIdx());
```
Comment (d-b-w) [inline Code/GraphMol/FileParsers/MolFileWriter.cpp:403]: huh.

File: Code/GraphMol/FileParsers/MolFileWriter.cpp (hunk 1 above)
Comment (greglandrum) [reply Code/GraphMol/FileParsers/MolFileWriter.cpp:1386]: Nope, that's intended to fail if you have a molecule with aromatic atoms and no bonds: that clearly cannot be kekulized.

Comment (d-b-w) [reply Code/GraphMol/FileParsers/MolFileWriter.cpp:1386]: cool

**Pull request merged**